### DISCUSSION TITLE
WindowTop alternative suggestiongs

### DISCUSSION THREAD

**Discussion by still-dreaming-1** [WindowTop/WindowTop-App#271] [Q&A] [answered]
Can anyone suggest alternatives to WindowTop that are less buggy? I really like and somewhat need a program like this, but it is extremely buggy. All the settings I have tried either do the opposite of what they say or they work sometimes and not others. The primary features of being able to show a window that is transparent, click through, and on top of all other windows (can always be seen behind them), is working worse and worse. At first it basically worked, but only if you know how to work around the bugs. Now I can't seem to get it to work with anything other than one other window. I understand software development is hard and users need to report bugs and be patient, but there should still be some level of already existing quality, especially for a product you are expected to pay for. I am not going to report all these issues and wait for them to be fixed when the quality is this bad, I need an alternative.

UPDATE:
Apparently asking for alternatives is considered crying, even after I found one and accepted it as the answer. The maintainer of this is the one who was crying and saying they can't guess what my bugs are even though I never asked them to fix the bugs or guess what they were, I was just responding to their cry and so they accused me of crying and blocked me from replying again.

**Comment by gileli121** [comment]
@still-dreaming-1 

Let's get to the point; currently, you will not find any alternative for the "Glass" (smart transparency) feature and not for the dark mode feature, Anchors feature, and probably also not for the PiP feature (and maybe I am missing a few more features).
You have much more chances to get what you want if you report the bugs here and the steps to reproduce them so I can fix them instead of waiting for someone to write from scratch something extremely hard to develop and will take a lot of time to make.
If the price hurts you, we can talk about it. I understand your point.
I can give you a special price if you report here bugs with good information about how to reproduce them.

By the way, if you claim that the app is extremely buggy, I expect to see it in its ratings, and I expect more user reports via the contact form. This is not the case. The app rating in Microsoft Store is 4.7-4.8 (of 5.0 maximum rating) according to 495 users.
When there is a very buggy version, I get much more reports about it. I don't see it in the last version (5.19.1). I did see it, for example, in 5.17

**Comment by still-dreaming-1** [comment, marked as answer]
I just spend an hour and a half learning AutoHotKey and working up a script to do what I needed. It works well enough and is a slightly better solution for me than WindowTop, but I can now understand better why it is difficult to get a program like this to work correctly. It seems many programs fight against these window changes in different ways under various circumstances. I'm having slightly better luck using Firefox as my behind window than Chrome. I'm currently liking this solution slightly better than WindowTop for my use case because if things get wonky, I just rerun the script and it goes back to how I want. Keep in mind that in AutoHotKey, `;` means starting a code comment.

```
#NoEnv ; Recommended for performance and compatibility with future AutoHotkey releases.
; #Warn ; Enable warnings to assist with detecting common errors.
SendMode Input ; Recommended for new scripts due to its superior speed and reliability.
SetWorkingDir %A_ScriptDir% ; Ensures a consistent starting directory.
; Everything up to this point was auto generated by creating a new AutoHotKey script

; Capture the window of your choice, that you want to display behind the other windows,
; into a variable called behindWindow. "Cab ride" is the partial text of the window
; title I need to display behind other windows. Change it to the title of the window
; you want to display behind.
WinGet, behindWindow, ID, Cab ride

; Set window transparency to about 17% visible (43 is about 17% of 255)
WinSet, Transparent, 43, ahk_id %behindWindow%

; Set window to be always on top
WinSet, AlwaysOnTop, On, ahk_id %behindWindow%

; Enable click-through
WinSet, ExStyle, +0x80020, ahk_id %behindWindow%

; Disable click-through
;WinSet, ExStyle, -0x20, ahk_id %behindWindow%
```

**Reply by gileli121** [reply]
It is extremely rare that someone reports that one of these features is not working as expected.
Maybe you expected a different behavior and did not see any bugs.
I can't tell what was wrong by guessing.

Anyway, great that you found your solution.

**Reply by still-dreaming-1** [reply]
I wasn't asking for anything to be fixed or for anyone to guess what is wrong with the software. I was asking for an alternative because it was so buggy that it didn't even make sense to try to report the bugs, so I was asking for alternative software suggestions. It always has been buggy, for many years, but not enough to make me stop using it until recently. I have grown weary of reporting most bugs for most programs because most programs contain lots of bugs and I can't spend all my time just reporting bugs that usually end up not getting fixed anyways. It is true there is a possibility some or all of the things I experienced were not really bugs and it was actually working as intended, but that would be an indication of something even worse than bugs.

**Reply by gileli121** [reply]
I have no idea about the bugs you get because you don't want to invest a small effort in reporting a single bug example.
Sounds like you don't want to spend your time, but it is precisely what you are doing here in writing these long messages.
I expect to see more reports and sales decrease if what you claim is correct, and I don't see any of these signs.
So again, no idea about what exactly your experience but it looks like you already decided to search for an alternative, and in such a case, this is not the right place to do it.

We could talk about some deals like making you a much better price, and you will report the issues, so I will try to help you, but you have already decided to continue crying here and wasting our time.

This discussion is locked to save my time and yours, as you wanted.
Go ahead and search for alternatives in another place.
